stacker/dag/__init__.py

```diff
@@ -0,0 +1,313 @@
+import threading
+import logging
+from copy import copy, deepcopy
+from collections import deque
+
+logger = logging.getLogger(__name__)
+
+try:
+    from collections import OrderedDict
+except:
+    from ordereddict import OrderedDict
+
+
+class DAGValidationError(Exception):
+    pass
+
+
+class DAG(object):
```

Comment:
Even though we plan to pull this in, do you think a PR to the origin would be worth while with the additional `walk` and `transpose` methods?